enumerable_project.rb

```diff
@@ -0,0 +1,139 @@
+ Copyright (c) Muhammasd Arslan
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    return self unless block_given?
+
+    arr = []
+    my_each do |x|
+      arr << x if yield(x) == true
+    end
+    arr
+  end
+
+  def my_all?
+    return self unless block_given?
+
+    my_each do |x|
+      return false if yield(x) == false
+    end
+    true
+  end
```

Comment:
my_all?

when no block or argument is given returns true when none of the collection members are false or nil:
`p [1, true, 'hi', []].my_all?` should return true.

when a class is passed as an argument returns true if all of the collection is a member of such class
`p [1,1,1].my_all?(Integer)` should return true

when a Regex is passed as an argument returns true if all of the collection matches the Regex
`p ['d','d'].my_all?(/d/)` should return true

when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`p [3,3,3,3].my_all?(3)` should return true